.github/workflows/nightly-build.yml

```diff
@@ -0,0 +1,97 @@
+name: Nightly Build
+
+on:
+  schedule:
+    - cron: '0 20 * * *'
+jobs:
+  nightly-build:
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: [ "amd64","arm64" ]
+        container: [ "rhel8:latest", "rhel9:latest", "ubuntu2204:latest" ]
+        include:
+          - container: "rhel8:latest"
+            cmd: "make rpm"
+            distro_symbol: "el8"
+            ext: "rpm"
+          - container: "rhel9:latest"
+            cmd: "make rpm"
+            distro_symbol: "el9"
+            ext: "rpm"
+          - container: "ubuntu2204:latest"
+            cmd: "make deb"
+            distro_symbol: "debian"
+            ext: "deb"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+      - name: Check if the last commit was today
+        id: check-commit
+        run: |
+          LAST_COMMIT_DATE="$(git log -1 --format=%cd --date=short)"
+          TODAY="$(date -u +%Y-%m-%d)"
+          if [ "$LAST_COMMIT_DATE" == "$TODAY" ]; then
+            echo "commit_today=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "commit_today=false" >> "$GITHUB_OUTPUT"
+          fi
```

Comment:
You can extract those steps into a separate job and have a single if condition for the `nightly-build` job.
You will need to check out the code in both of them.